### PR TITLE
[PREGEL] Update pregel status documentation

### PR DESCRIPTION
**Pull request by jvolmer**
- Update timings
- Add details description

### CODE REVIEW

File: Documentation/DocuBlocks/Rest/Pregel/get_api_control_pregel_status.md [hunk 1]
```diff
@@ -85,6 +87,44 @@ Total number of vertices processed.
...
+
+@RESTSTRUCT{messagesSent,get_api_control_pregel_detail_allgssstatus,number,optional,int64}
+Number of messages sent in this step.
+
+@RESTSTRUCT{messagesReceived,get_api_control_pregel_detail_allgssstatus,number,optional,int64}
+Number of messages received in this step.
+
+@RESTSTRUCT{memoryBytesUsedForMessages,get_api_control_pregel_detail_allgssstatus,number,optional,int64}
+Number of bytes used in memory for the messages in this step.
+
```
Comment (jvolmer) [inline Documentation/DocuBlocks/Rest/Pregel/get_api_control_pregel_status.md:127]: I'm not sure if this code is the right way to describe the deeply nested 'detail' construct, this is an example how 'detail' can look like:
```
    "detail" : { 
      "aggregatedStatus" : { 
        "timeStamp" : "2022-07-25T08:49:00Z", 
        "graphStoreStatus" : { 
          "verticesLoaded" : 10, 
          "edgesLoaded" : 17, 
          "memoryUsed" : 1375, 
          "verticesStored" : 10 
        }, 
        "allGssStatus" : { 
          "items" : [ 
            { 
              "verticesProcessed" : 10, 
              "messagesSent" : 17, 
              "messagesReceived" : 0, 
              "memoryBytesUsedForMessages" : 816 
            }, 
            { 
              "verticesProcessed" : 10, 
              "messagesSent" : 27, 
              "messagesReceived" : 17, 
              "memoryBytesUsedForMessages" : 2112 
            }, 
            { 
              "verticesProcessed" : 10, 
              "messagesSent" : 33, 
              "messagesReceived" : 27, 
              "memoryBytesUsedForMessages" : 2880 
            }, 
            { 
              "verticesProcessed" : 10, 
              "messagesSent" : 27, 
              "messagesReceived" : 33, 
              "memoryBytesUsedForMessages" : 2880 
            }, 
            { 
              "verticesProcessed" : 10, 
              "messagesSent" : 0, 
              "messagesReceived" : 27, 
              "memoryBytesUsedForMessages" : 1296 
            } 
          ] 
        } 
      }, 
      "workerStatus" : { 
        "PRMR-d0101d1c-afc4-4b92-b7a5-2d5521a2a5ba" : { 
          "timeStamp" : "2022-07-25T08:49:00Z", 
          "graphStoreStatus" : { 
            "verticesLoaded" : 1, 
            "edgesLoaded" : 4, 
            "memoryUsed" : 201, 
            "verticesStored" : 1 
          }, 
          "allGssStatus" : { 
            "items" : [ 
              { 
                "verticesProcessed" : 1, 
                "messagesSent" : 4, 
                "messagesReceived" : 0, 
                "memoryBytesUsedForMessages" : 192 
              }, 
              { 
                "verticesProcessed" : 1, 
                "messagesSent" : 7, 
                "messagesReceived" : 3, 
                "memoryBytesUsedForMessages" : 480 
              }, 
              { 
                "verticesProcessed" : 1, 
                "messagesSent" : 7, 
                "messagesReceived" : 6, 
                "memoryBytesUsedForMessages" : 624 
              }, 
              { 
                "verticesProcessed" : 1, 
                "messagesSent" : 7, 
                "messagesReceived" : 7, 
                "memoryBytesUsedForMessages" : 672 
              }, 
              { 
                "verticesProcessed" : 1, 
                "messagesSent" : 0, 
                "messagesReceived" : 7, 
                "memoryBytesUsedForMessages" : 336 
              } 
            ] 
          } 
        }, 
        "PRMR-79f1ec4a-98e7-4169-9cc8-6c428873a41d" : { 
          "timeStamp" : "2022-07-25T08:49:00Z", 
          "graphStoreStatus" : { 
            "verticesLoaded" : 5, 
            "edgesLoaded" : 5, 
            "memoryUsed" : 592, 
            "verticesStored" : 5 
          }, 
          "allGssStatus" : { 
            "items" : [ 
              { 
                "verticesProcessed" : 5, 
                "messagesSent" : 5, 
                "messagesReceived" : 0, 
                "memoryBytesUsedForMessages" : 240 
              }, 
              { 
                "verticesProcessed" : 5, 
                "messagesSent" : 9, 
                "messagesReceived" : 6, 
                "memoryBytesUsedForMessages" : 720 
              }, 
              { 
                "verticesProcessed" : 5, 
                "messagesSent" : 10, 
                "messagesReceived" : 10, 
                "memoryBytesUsedForMessages" : 960 
              }, 
              { 
                "verticesProcessed" : 5, 
                "messagesSent" : 9, 
                "messagesReceived" : 11, 
                "memoryBytesUsedForMessages" : 960 
              }, 
              { 
                "verticesProcessed" : 5, 
                "messagesSent" : 0, 
                "messagesReceived" : 9, 
                "memoryBytesUsedForMessages" : 432 
              } 
            ] 
          } 
        }, 
        "PRMR-4ed0c91d-2362-4d6b-8499-504e3f41ca1e" : { 
          "timeStamp" : "2022-07-25T08:49:00Z", 
          "graphStoreStatus" : { 
            "verticesLoaded" : 4, 
            "edgesLoaded" : 8, 
            "memoryUsed" : 582, 
            "verticesStored" : 4 
          }, 
          "allGssStatus" : { 
            "items" : [ 
              { 
                "verticesProcessed" : 4, 
                "messagesSent" : 8, 
                "messagesReceived" : 0, 
                "memoryBytesUsedForMessages" : 384 
              }, 
              { 
                "verticesProcessed" : 4, 
                "messagesSent" : 11, 
                "messagesReceived" : 8, 
                "memoryBytesUsedForMessages" : 912 
              }, 
              { 
                "verticesProcessed" : 4, 
                "messagesSent" : 16, 
                "messagesReceived" : 11, 
                "memoryBytesUsedForMessages" : 1296 
              }, 
              { 
                "verticesProcessed" : 4, 
                "messagesSent" : 11, 
                "messagesReceived" : 15, 
                "memoryBytesUsedForMessages" : 1248 
              }, 
              { 
                "verticesProcessed" : 4, 
                "messagesSent" : 0, 
                "messagesReceived" : 11, 
                "memoryBytesUsedForMessages" : 528 
              } 
            ] 
          } 
        } 
      } 
    } 
```

**Comment by jvolmer** [comment]
TODO rename memoryUsed to memoryBytesUsed (is done in PR https://github.com/arangodb/arangodb/pull/16634)

File: Documentation/DocuBlocks/Rest/Pregel/get_api_control_pregel_status.md (hunk 1 above)
Comment (Simran-B) [reply Documentation/DocuBlocks/Rest/Pregel/get_api_control_pregel_status.md:138]: `@RESTREPLYBODY` is for the top-level attributes only. All nested objects and arrays need to use `@RESTSTRUCT`.
I'm not aware of a way to describe the objects with DB-Server IDs as keys using our DocuBlock system, even though this is possible in Swagger. As a workaround, the `workerStatus` can refer to the `aggregatedStatus` attribute to describe the response structure.

**Review by Simran-B — APPROVED** [review]

**Comment by ansoboleva** [comment]
@jvolmer is it for 3.10 or devel (or both?) The milestone is set to 3.10, but the target branch is devel. In the accompanying docs PR, 3.10 files are updated. Please clarify the affected version(s).

**Comment by Simran-B** [comment]
The code changes were merged into devel before the 3.10 branch-off.
Backport of this PR to 3.10: #16759

**Comment by jvolmer** [comment]
Thanks for the backport, Simran! I was really unsure whether the added RestStructs in get_api_control_pregel_status.md are correct as they are, but I think you figured it out?

**Pull request merged**